db/model/archival/archivalDb.ts

```diff
@@ -788,10 +889,167 @@ export const getGdocPostChecksumsFromDb = async (
 
     if (gdocPosts.length === 0) return []
 
-    // Build result with checksums and hashed checksums
+    const postIds = gdocPosts.map((p) => p.id)
+    const linkedCharts = await getLinkedChartsFromPosts(knex, postIds)
+    const chartsByPost = groupLinkedChartsByPost(gdocPosts, linkedCharts)
```

Comment:
I'm not entirely sure, but it could be that we need to also take care of chart redirects in here... :/

As in, the slug specified in `posts_gdocs_links` may not be current any more, but needs to be resolved using a redirect.